apps/backend/core/configuration/validation/base.py

```
@@ -0,0 +1,279 @@
+    latency_ms: float | None = Field(default=None, description="Validation latency in milliseconds")
+    timestamp: datetime = Field(default_factory=datetime.utcnow, description="When validation occurred")
+    metadata: dict[str, Any] = Field(default_factory=dict, description="Additional validation metadata")
+
+    class Config:
+        """Pydantic model configuration."""
+
+        json_encoders = {
+            datetime: lambda v: v.isoformat(),
+            ValidationStatus: lambda v: v.value,
+            ValidationLevel: lambda v: v.value,
+        }
```

Ruff (0.14.4)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)